### ISSUE TITLE
Vulkan backend on m1 Mac ( device does not support wide lines)

### ISSUE THREAD

**Issue by Linyou** [taichi-dev/taichi#3416]
The problem was occurred when running ggui_example/mass_spring_game_ggui.py

```
❯ python mass_spring_game_ggui.py                                          
[Taichi] version 0.8.4, llvm 10.0.0, commit f665c28c, osx, python 3.9.7
[Taichi] Starting on arch=vulkan
[W 11/07/21 22:14:21.787 427294] [embedded_device.cpp:create_logical_device@407] Potential non-conformant Vulkan implementation, enabling VK_KHR_portability_subset
[W 11/07/21 22:14:21.787 427294] [embedded_device.cpp:create_logical_device@464] Taichi GPU GUI requires wide lines support
[W 11/07/21 22:14:21.986 427294] [vulkan_device.cpp:buffer@527] Overriding last binding
[W 11/07/21 22:14:21.991 427294] [vulkan_device.cpp:buffer@527] Overriding last binding
[mvk-error] VK_ERROR_FEATURE_NOT_PRESENT: vkCmdSetLineWidth(): The current device does not support wide lines.
[E 11/07/21 22:14:22.000 427294] [vulkan_device.cpp:submit_synced@1387] Vulkan Error : -8 : failed to submit command buffer
```

```
                            * Taichi Core - Stack Traceback *                             
==========================================================================================
|                       Module |  Offset | Function                                      |
|----------------------------------------------------------------------------------------|
*               taichi_core.so |     120 | taichi::Logger::error(std::__1::basic_string< |
                                         | char, std::__1::char_traits<char>, std::__1:: |
                                         | allocator<char> > const&, bool)               |
*               taichi_core.so |     540 | taichi::lang::vulkan::VulkanStream::submit_sy |
                                         | nced(taichi::lang::CommandList*)              |
*               taichi_core.so |     508 | taichi::ui::vulkan::Renderer::draw_frame(taic |
                                         | hi::ui::vulkan::Gui*)                         |
*               taichi_core.so |      32 | taichi::ui::vulkan::Window::show()            |
*               taichi_core.so |     108 | void pybind11::cpp_function::initialize<pybin |
                                         | d11::cpp_function::cpp_function<void, taichi: |
                                         | :ui::PyWindow, pybind11::name, pybind11::is_m |
                                         | ethod, pybind11::sibling>(void (taichi::ui::P |
                                         | yWindow::*)(), pybind11::name const&, pybind1 |
                                         | 1::is_method const&, pybind11::sibling const& |
                                         | )::'lambda'(taichi::ui::PyWindow*), void, tai |
                                         | chi::ui::PyWindow*, pybind11::name, pybind11: |
                                         | :is_method, pybind11::sibling>(void&&, taichi |
                                         | ::ui::PyWindow (*)(), pybind11::name const&,  |
                                         | pybind11::is_method const&, pybind11::sibling |
                                         |  const&)::'lambda'(pybind11::detail::function |
                                         | _call&)::__invoke(pybind11::detail::function_ |
                                         | call&)                                        |
*               taichi_core.so |    3316 | pybind11::cpp_function::dispatcher(_object*,  |
                                         | _object*, _object*)                           |
*                    python3.9 |      80 | (null)                                        |
*                    python3.9 |     340 | (null)                                        |
*                    python3.9 |     268 | (null)                                        |
*                    python3.9 |     572 | (null)                                        |
*                    python3.9 |   29268 | (null)                                        |
*                    python3.9 |     116 | (null)                                        |
*                    python3.9 |     572 | (null)                                        |
*                    python3.9 |   29396 | (null)                                        |
*                    python3.9 |    2968 | (null)                                        |
*                    python3.9 |     376 | (null)                                        |
*                    python3.9 |     816 | (null)                                        |
*                    python3.9 |    2916 | (null)                                        |
*                    python3.9 |    1272 | (null)                                        |
*                    python3.9 |      56 | (null)                                        |
*                         dyld |     520 | (null)                                        |
==========================================================================================
```
When I remove the 158 line, the code run normally, but can not display any line

https://github.com/taichi-dev/taichi/blob/0e4ca37c2581f6e449c193612e77657c4dd87865/examples/ggui_examples/mass_spring_game_ggui.py#L158

Currently, it seem like the Vulkan backend on Mac, dose not support wide lines,  is there have any way to work around this problem?  maybe replace canvas.lines() with another function?

**Comment by ailzhang** [comment]
cc: @AmesingFlank ?

**Comment by bobcao3** [comment]
Fixed in #3432. Variable width line is not supported yet, but since that's a different issue, closing this one for now.
